pystac_client/warnings.py

```diff
@@ -0,0 +1,81 @@
+from contextlib import contextmanager
+from typing import Iterator, Union
+import warnings
+
+from pystac_client.conformance import ConformanceClasses
+
+
+FALLBACK_MSG = "Falling back to pystac. This might be slow."
+
+
+def DOES_NOT_CONFORM_TO(conformance_class: Union[str, ConformanceClasses]) -> str:
+    return f"Server does not conform to {conformance_class}"
+
+
+class PystacClientWarning(UserWarning):
+    """Base warning class"""
+
+    ...
+
+
+class NoConformsTo(PystacClientWarning):
+    """Inform user when client does not have "conformsTo" set"""
+
+    ...
+
+
+class DoesNotConformTo(PystacClientWarning):
+    """Inform user when client does not conform to extension"""
+
+    ...
+
+
+class MissingLink(PystacClientWarning):
+    """Inform user when link is not found"""
+
+    ...
+
+
+class FallbackToPystac(PystacClientWarning):
+    """Inform user when falling back to pystac implementation"""
+
+    ...
+
+
+@contextmanager
+def strict() -> Iterator[None]:
```

Comment:
This could accept optional categories if that seems helpful.

Reply:
My first thought is "no" -- either you use `strict`, or you build your own behavior by some other filter mechanism?